pyqir-generator/tests/test_external_functions.py

```diff
@@ -128,11 +137,11 @@ def test_call_two_dynamic_results(self) -> None:
         )
 
     def test_call_numbers(self) -> None:
-        b = True
+        b = Value.integer(types.BOOL, True)
```

Comment:
Is it necessary to do this ? It looks like an overcomplicated and confusing syntax to me. Would defining a `bool` field in `Value` make it easier to use ?

Reply:
We could define `Value.bool` as a shorthand for creating an `i1` value.

In general, PyQIR needs a way to create values of any integer type. For example `Value.integer(types.Integer(16), 1)` for a 16-bit integer. Would you prefer to have more concise syntax for all integer values in addition to bools?

Reply:
Well, that is a matter of taste but in general I am in favour of conciseness. I would assume that passing the width directly to `Value.integer` for any non default 32-bit integer would make it easier to use.

Reply:
Something to consider is that we're planning in the near future to transition PyQIR from being a deep embedding of LLVM to a shallow one. At that point, we won't be able to create constant values without a reference to the LLVM context (which all types have). We're trying to design the current API so that this switch over will cause as few breaking changes as possible. So I'd prefer to keep requiring a type object for constant value creation.

I took another stab at making the syntax more concise given this constraint, though. The new syntax is:

```
const(types.BOOL, True)
const(types.Int(16), 123)
```

Reply:
Looks good to me !